services/userServices.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-undef */
+import { userDao } from '../models';
+import bcrypt from 'bcrypt';
+
+const signUp = async (name, email, user_account, phone_number, password) => {
+  const saltRounds = 10;
+  const salt = await bcrypt.genSalt(saltRounds);
+  const hashedPw = await bcrypt.hash(password, salt);
+
+  const existingUser = await userDao.getUser(user_account);
+
+  if (existingUser.length) {
+    const error = new Error('error');
+    error.statusCode = 409;
+    throw error;
+  }
+
```

Comment:
```suggestion

const signUp = async (name, email, user_account, phone_number, password) => {
  const existingUser = await userDao.getUser(user_account);

  if (existingUser.length) {
    const error = new Error('error');
    error.statusCode = 409;
    throw error;
  }

  const saltRounds = 10;
  const salt = await bcrypt.genSalt(saltRounds);
  const hashedPw = await bcrypt.hash(password, salt);
```
미정님, 코드 순서를 변경하셔서 이메일이 존재하는지를 먼저 체크하면 좋을 것 같습니다 👍
제가 왜 이 방법을 추천하는 것일까요?

Reply:
저의 개인적인 .. 직감과 추측과 코드를 몇번 돌려본 결과 체감상 먼저 중복 아이디를 체크하는것이 먼저 비밀번호를 해싱하는 것 보다 빠르게 느껴졌습니다! 
- 기존의 코드처럼 먼저 중복 아이디를 체크하지 않았을 경우에는 일단 사용자가 입력한 비밀번호를 냅다 해싱→그 뒤에 아이디를 중복 체크하는 과정이기 때문에 효율이 떨어집니다
- 아이디 중복 체크를 우선적으로 한다면 비밀번호를 해싱하는 과정 이전에 이미 존재하는 아이디를 입력했을 경우에는 바로 에러를 리턴하기 때문에 해싱하는 과정을 거치지 않아도 되서 더 .. 싸게 칩니다.

Reply:
ㅋㅋㅋㅋ 미정님 너무 좋은 생각입니다 싸게 치이는 것 맞습니다.
필요없을때는 냅다 해싱 안해도 되니까요 :) 이용자가 있을때만 해싱하면 좋을 것 같습니다.

속도를 체감으로 느끼셨다니 이건 너무 대단한데요 👁️👃👁️